src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -144,27 +155,49 @@ CHIP_ERROR GeneralCommissioningAttrAccess::ReadSupportsConcurrentConnection(Attr
     return aEncoder.Encode(supportsConcurrentConnection);
 }
 
-} // anonymous namespace
+void GeneralCommissioningAttrAccess::InvokeCommand(HandlerContext & handlerContext)
+{
+    ChipLogDetail(Zcl, "GeneralCommissioningAttrAccess: InvokeCommand");
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Commands::ArmFailSafe::Id:
+        ChipLogDetail(Zcl, "GeneralCommissioningAttrAccess: Entering ArmFailSafe");
+
+        CommandHandlerInterface::HandleCommand<Commands::ArmFailSafe::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleArmFailSafe(ctx, req); });
+        break;
+
+    case Commands::SetRegulatoryConfig::Id:
+        ChipLogDetail(Zcl, "GeneralCommissioningAttrAccess: Entering SetRegulatoryConfig");
+
+        CommandHandlerInterface::HandleCommand<Commands::SetRegulatoryConfig::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleSetRegulatoryConfig(ctx, req); });
+        break;
+
+    case Commands::CommissioningComplete::Id:
+        ChipLogDetail(Zcl, "GeneralCommissioningAttrAccess: Entering CommissioningComplete");
+
+        CommandHandlerInterface::HandleCommand<Commands::CommissioningComplete::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleCommissioningComplete(ctx, req); });
+        break;
+    }
+}
 
-bool emberAfGeneralCommissioningClusterArmFailSafeCallback(app::CommandHandler * commandObj,
-                                                           const app::ConcreteCommandPath & commandPath,
-                                                           const Commands::ArmFailSafe::DecodableType & commandData)
+void GeneralCommissioningAttrAccess::HandleArmFailSafe(HandlerContext & ctx, const Commands::ArmFailSafe::DecodableType & req)
 {
     MATTER_TRACE_SCOPE("ArmFailSafe", "GeneralCommissioning");
     auto & failSafeContext = Server::GetInstance().GetFailSafeContext();
     Commands::ArmFailSafeResponse::Type response;
 
-    ChipLogProgress(FailSafe, "GeneralCommissioning: Received ArmFailSafe (%us)",
-                    static_cast<unsigned>(commandData.expiryLengthSeconds));
+    ChipLogProgress(FailSafe, "GeneralCommissioning: Received ArmFailSafe (%us)", static_cast<unsigned>(req.expiryLengthSeconds));
 
     /*
      * If the fail-safe timer is not fully disarmed, don't allow arming a new fail-safe.
      * If the fail-safe timer was not currently armed, then the fail-safe timer SHALL be armed.
      * If the fail-safe timer was currently armed, and current accessing fabric matches the fail-safe
      * context’s Fabric Index, then the fail-safe timer SHALL be re-armed.
      */
-
-    FabricIndex accessingFabricIndex = commandObj->GetAccessingFabricIndex();
+    FabricIndex accessingFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
```

Comment:
Could also define commandObj up at the top of the function to avoid changing all this code.  But either way.  I can see the benefit of doing this change.

Reply:
commandObj is a pointer to ctx.mCommandHandler, to avoid changing the original function, we need to

app::CommanHandler * commandObj = &ctx.mCommandHandler;

I think using ctx.mCommandHandler directly is more intuitive